cpu/atmega_common/include/atmega_gpio.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2015 HAW Hamburg
+ *               2016 INRIA
+
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_atmega_common
+ * @ingroup     drivers_periph_gpio
+ * @{
+ *
+ * @file
+ * @brief       Macros and inline functions for accessing GPIOs of the ATmega family
+ *
+ * @author      René Herthel <rene-herthel@outlook.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ * @author      Laurent Navet <laurent.navet@gmail.com>
+ */
+
+#ifndef ATMEGA_GPIO_H
+#define ATMEGA_GPIO_H
+#include <stdio.h>
+
+#include <avr/interrupt.h>
+
+#include "cpu.h"
+#include "board.h"
+#include "periph/gpio.h"
+#include "periph_conf.h"
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define GPIO_BASE_PORT_A        (0x20)
+#define GPIO_OFFSET_PORT_H      (0xCB)
+#define GPIO_OFFSET_PIN_PORT    (0x02)
+#define GPIO_OFFSET_PIN_PIN     (0x03)
+
+/**
+ * @brief     Extract the pin number of the given pin
+ */
+static inline uint8_t _pin_num(gpio_t pin)
```

Comment:
Namespace collisions preeminent. If you publicize these functions they should be prefixed with something like `atmega_gpio`.

Reply:
Those are not public functions, but private ATmega related functions. A namespace collision would only occur if code would use ATmega peripheral GPIOs and non-ATmega peripheral GPIOs. But that would make little sense.

Reply:
Those are in a header, right? So they are publich *will* lead to collisions if another header included somewhere by chance have also something named e.g. `_pin_num()`. Another problem is that by moving those names to a header you violate the [C standard](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf):

> All identifiers that begin with an underscore are always reserved for use as identifierswith file scope in both the ordinary and tag name spaces. [page 182]

Reply:
OK, I will update the code as suggested

Reply:
Done

Reply:
> —  All identifiers that begin with an underscore and either an uppercase letter or anotherunderscore are always reserved for any use.
> —  All identifiers that begin with an underscore are always reserved for use as identifiers with file scope in both the ordinary and tag name spaces

I thought a bit about the meaning of these.

I understand the first as `__foo`, `_FOO`, or `__FOO` is reserved; it must not be used. (To my understanding those are reserved to the compiler.)

I understand the second as `_foo` is reserved in file scope; it must not be used until the scope is limited (e.g. a variable in a function is fine, or a member of a `struct` or `union` would be fine).

If this interpretation is correct, then function names with leading underscores shouldn't be used at all. I can find some discussions at stackoverflow that seem to support this interpretation, but nothing with enough authority that I feel sure about the way I read the standard here.

In any case: Thanks for insisting on me fixing this!